cfme/automate/provisioning_dialogs.py

```diff
@@ -1,138 +1,238 @@
+    class form(ProvDiagForm):  # noqa
+        add = Button('Add')
 
-    pretty_attrs = ['name', 'description', 'content']
 
-    def __init__(self, type, name=None, description=None, content=None, appliance=None):
+class ProvDiagEditView(ProvDiagView):
+    @property
+    def is_displayed(self):
+        # FIXME https://github.com/ManageIQ/manageiq-ui-classic/issues/1983
+        # 'Editing' should only be in the title once
+        expected_title = 'Editing Editing Dialog "{}"'.format(self.context['object'].description)
+        return (
+            self.in_customization and
+            self.title.text == expected_title and
+            self.form.is_displayed)
+
+    title = Text('#explorer_title_text')
+
+    @View.nested
+    class form(ProvDiagForm):  # noqa
+        save = Button('Save')
+        reset = Button('Reset')
+
+
+class ProvisioningDialog(Updateable, Pretty, Navigatable):
+    HOST_PROVISION = 'Host Provision'
+    VM_MIGRATE = 'VM Migrate'
+    VM_PROVISION = 'VM Provision'
+    SYSTEM_PROVISION = 'Configured System Provision'
+    ALLOWED_TYPES = {HOST_PROVISION, VM_MIGRATE, VM_PROVISION, SYSTEM_PROVISION}
+
+    pretty_attrs = ['name', 'description', 'diag_type', 'content']
+
+    def __init__(self, diag_type, name=None, description=None, content=None, appliance=None):
         Navigatable.__init__(self, appliance=appliance)
         self.name = name
         self.description = description
-        self.type = type
         self.content = content
+        if diag_type in self.ALLOWED_TYPES:
+            self.diag_type = diag_type
+        else:
+            raise TypeError('Type must be one of ProvisioningDialog constants: {}'
+                            .format(self.ALLOWED_TYPES))
 
     def __str__(self):
         return self.name
 
     @property
-    def type_nav(self):
-        return self.type[0]
-
-    @property
-    def type_tree_nav(self):
-        return self.type[1]
-
-    @property
     def exists(self):
         try:
             navigate_to(self, 'Details')
-            return True
         except CandidateNotFound:
             return False
+        return True
 
     def create(self, cancel=False):
-        navigate_to(self, 'Add')
-        fill(self.form, self.__dict__, action=form_buttons.cancel if cancel else form_buttons.add)
-        flash.assert_no_errors()
+        view = navigate_to(self, 'Add')
+        # might not want to use pretty_attrs here, overloading its intended use
+        fill_args = {key: self.__dict__[key] for key in self.pretty_attrs}
+        view.form.fill(fill_args)
+        if cancel:
+            flash_msg = 'Add of new Dialog was cancelled by the user'
+            btn = view.form.cancel
+        else:
+            flash_msg = 'Dialog "{}" was added'.format(self.name)
+            btn = view.form.add
+
+        btn.click()
+        view = self.create_view(ProvDiagAllView if cancel else ProvDiagDetailsView)
+        assert view.is_displayed
+        view.flash.assert_no_error()
```

Comment:
**REQUIRED**
`flash.assert_no_error()` is a part of `flash.assert_success_message()`, so it's not needed.

Reply:
Keep forgetting that, good call.